packages/melos/lib/src/commands/bootstrap.dart

```diff
@@ -54,34 +56,44 @@ mixin _BootstrapMixin on _CleanMixin {
                 devDependencies: bootstrapCommandConfig.devDependencies,
               );
             }).drain<void>();
+
+            logger
+              ..child(successLabel, prefix: '> ')
+              ..newLine();
           }
 
-          await _linkPackagesWithPubspecOverrides(
-            workspace,
-            enforceLockfile: enforceLockfile,
-            noExample: noExample,
-          );
+          if (!skipLinking) {
+            logger.log(
+              'Running "$pubCommandForLogging" in workspace packages...',
+            );
+
+            await _linkPackagesWithPubspecOverrides(
+              workspace,
+              enforceLockfile: enforceLockfile,
+              noExample: noExample,
+            );
+
+            logger
+              ..child(successLabel, prefix: '> ')
+              ..newLine();
+          }
         } on BootstrapException catch (exception) {
           _logBootstrapException(exception, workspace);
           rethrow;
         }
 
-        logger.child(successLabel, prefix: '> ');
-
         if (workspace.config.ide.intelliJ.enabled) {
-          logger
-            ..newLine()
-            ..log('Generating IntelliJ IDE files...');
+          logger.log('Generating IntelliJ IDE files...');
 
           await cleanIntelliJ(workspace);
           await workspace.ide.intelliJ.generate();
-          logger.child(successLabel, prefix: '> ');
+          logger
+            ..child(successLabel, prefix: '> ')
+            ..newLine();
```

Comment:
What is the reasoning behind the `newLine` changes?

Reply:
Because the command `_linkPackagesWithPubspecOverrides ` is now conditional, we have to adapt to `success + newLine` for every step otherwise the outputs looks congested.

We also haven't changed the logger drastically, otherwise other tests could have failed.

Reply:
You can check the updated logs in the PR description above.